### PR TITLE
Added recenter button to direct nearby uploads

### LINKED ISSUE #1228: Add re-center button to Nearby map (in feature branch)
After testing the Nearby feature in `directNearbyUploads` branch for a bit, I really think that we need to provide a way for users to re-center the map on their current location. Users will be swiping a lot to switch between the list and the map - and if they swipe wrongly and move the map away, they have to reload Nearby or manually go back to their location.

I do recall that we discussed this somewhere else and it was mentioned that the Material guidelines recommend not having more than 1 FAB in the screen. However, I think that a small re-center icon on the upper right would not be amiss, especially given that our other FAB is actually in the bottom sheet and not on the map itself. For reference, Google Maps also uses 3 FABs, 1 of which is the re-center button.

Thoughts?

### PR DESCRIPTION
**Pull request by Jatin0312**
## Fixes issues #1228 added recenter button to nearby map 

Fixes #1228

Added a recenter fab button on the top right corner which redirects the camera to the current location.

## Screenshots showing what changed
 

bug2solved

### CODE REVIEW

**Comment by Jatin0312** [comment]
Since directNearbyUploads branche is not active, so should I make this PR to directNearByUploadsNew

**Comment by neslihanturan** [comment]
Yes It seems like a good idea

**Comment by Jatin0312** [comment]
Ok I will make a PR to directNearByUploadsNew.

File: build.gradle [hunk 1]
```diff
@@ -7,7 +7,7 @@ buildscript {
         google()
     }
     dependencies {
-        classpath "com.android.tools.build:gradle:${project.gradleVersion}"
```
Comment (maskaravivek) [inline build.gradle]: why this change?

Comment (Jatin0312) [reply build.gradle]: Actually they are one and the same, but still I will revert it and see why the build is failing.

**Comment by Jatin0312** [comment]
I think the build is failing since there is no constructor in NearbyAdapterFactory which takes null argument.
```
/home/travis/build/commons-app/apps-android-commons/app/src/test/java/fr/free/nrw/commons/nearby/NearbyAdapterFactoryTest.java:41: error: no suitable constructor found for NearbyAdapterFactory(<null>)
        NearbyAdapterFactory testObject = new NearbyAdapterFactory(null);
                                          ^
    constructor NearbyAdapterFactory.NearbyAdapterFactory() is not applicable
      (actual and formal argument lists differ in length)
    constructor NearbyAdapterFactory.NearbyAdapterFactory(Fragment,ContributionController) is not applicable
      (actual and formal argument lists differ in length)
/home/travis/build/commons-app/apps-android-commons/app/src/test/java/fr/free/nrw/commons/nearby/NearbyAdapterFactoryTest.java:49: error: no suitable constructor found for NearbyAdapterFactory(<null>)
        NearbyAdapterFactory testObject = new NearbyAdapterFactory(null);
```

**Comment by neslihanturan** [comment]
Currently I want to make this merge before #1241 to prevent you solve conflicts again. It would be great if you can solve Viveks review as fast as possible @Jatin0312 :)

**Comment by misaochan** [comment]
Yes, tests were already failing on this branch, it is expected since it was a large overhaul (and part of the reason I didn't want to merge to master yet initially ;)). I would suggest merging this PR, and then we can create an issue for "Rewrite tests for Nearby" or similar.

**Comment by Jatin0312** [comment]
Ok I will do it

**Comment by Jatin0312** [comment]
@neslihanturan made the requested changes.
Sorry for the delay.

**Comment by misaochan** [comment]
Looks good to me. @neslihanturan @maskaravivek do we want this in directNearbyUploads, or should we switch the base to directNearbyUploadsNew ?

Sorry for the delay @Jatin0312 as we made a few changes to the branches a couple of days ago.

**Comment by Jatin0312** [comment]
No problem @misaochan. :)

**Comment by neslihanturan** [comment]
It should be in directNearbyUploadsNew, besides I have tested it. Weirdly it gave me Caused by: ```
java.lang.NullPointerException
                                                                               at fr.free.nrw.commons.nearby.NearbyMapFragment.setListeners(NearbyMapFragment.java:214)
                                                                               at fr.free.nrw.commons.nearby.NearbyMapFragment.onCreate(NearbyMapFragment.java:107) 
```
On API Level 19, trying to understand the reason.

**Comment by Jatin0312** [comment]
Sorry I was busy with some work . I will look into this bug

**Comment by Jatin0312** [comment]
Checked on Android 4.4 API 19 was working fine.

**Comment by Jatin0312** [comment]
Should I make this PR to other branch if it's working fine.

**Comment by misaochan** [comment]
Hi @Jatin0312 , we are just waiting for Neslihan to re-test your work. Please be patient, as she has other tasks to do as well. :)

**Comment by neslihanturan** [comment]
Solved, thanks @Jatin0312 !

**Comment by Jatin0312** [comment]
@misaochan Really sorry I will not repeat the same mistake.
Thanks :)
